src/platform_impl/linux/event_loop.rs

```diff
@@ -421,6 +427,33 @@ impl<T: 'static> EventLoop<T> {
                 | EventMask::SCROLL_MASK,
             );
 
+            const LMB: u32 = 1;
+            // Wayland add Events
+            if is_wayland {
+              window.connect_button_press_event(move |window, event| {
```

Comment:
instead of detecting click in the whole window, can't we just attach this event on the `HeaderBar` itself when creating it, in `wayland/display.rs`

Reply:
In `header.rs`, I moved the header drag event to `Self::connect_move_window(&header, &window);`.